Classes/New Issue/IssueTemplates.swift

```diff
@@ -0,0 +1,233 @@
+//
+//  IssueTemplates.swift
+//  Freetime
+//
+//  Created by Ehud Adler on 11/3/18.
+//  Copyright © 2018 Ryan Nystrom. All rights reserved.
+//
+
+import Foundation
+import GitHubAPI
+import GitHubSession
+import Squawk
+
+struct IssueTemplate {
+    let title: String
+    let template: String
+}
+
+struct IssueTemplateDetails {
+    let owner: String
+    let repo: String
+    let session: GitHubUserSession?
+    let viewController: UIViewController
+    weak var delegate: NewIssueTableViewControllerDelegate?
+}
+
+enum IssueTemplateHelper {
+
+
+    static private func matches(
+        regex: String,
+        text: String) -> [String] {
+        do {
+            let regex = try NSRegularExpression(pattern: regex)
+            let str = text as NSString
+            let results = regex.matches(
+                in: text,
+                range: NSMakeRange(0, str.length))
+            return results.map { str.substring(with: $0.range)}
+
+        } catch let error as NSError {
+            print("invalid regex: \(error.localizedDescription)")
+            return []
+        }
+    }
+
+    static func getNameAndDescription(fromTemplatefile file: String) -> (name: String?, about: String?) {
+        let names = IssueTemplateHelper.matches(regex: "(?<=name:).*", text: file)
+        let abouts = IssueTemplateHelper.matches(regex: "(?<=about:).*", text: file)
+        let name = names.first?.trimmingCharacters(in: .whitespaces)
+        let about = abouts.first?.trimmingCharacters(in: .whitespaces)
+        return (name, about)
+    }
+
+    static func showIssueAlert(
+        with templates: [IssueTemplate],
+        details: IssueTemplateDetails) -> UIAlertController {
+
+        let alertView = UIAlertController(
+            title: NSLocalizedString("New Issue", comment: ""),
+            message: NSLocalizedString("Choose Template", comment: ""),
+            preferredStyle: .alert
+        )
+
+        for template in templates {
+            alertView.addAction(
+                UIAlertAction(
+                    title: template.title,
+                    style: .default,
+                    handler: { _ in
+
+                        guard let viewController = NewIssueTableViewController.createWithTemplate(
+                            client: GithubClient(userSession: details.session),
+                            owner: details.owner,
+                            repo: details.repo,
+                            template: template.template,
+                            signature: template.title == "Bug Report" ? .bugReport : .sentWithGitHawk
```

Comment:
How does everyone feel about this check?

Is this essentially saying that if *anybody* creates a template in their repository called "Bug Report" then it will use the bug report template? Or is this limited just to the GitHawk repo?

I ask this because the Bug Report signature is pretty specific to GitHawk even going as far to include TestFlight information which isn't really applicable if it's a bug report on _pluck random repo out of the air_ equally the device info probably isn't that important.

Might be missing something here though!

Reply:
Ya that a good point. I meant to revisit this. Maybe check if repo =  GitHawk or owner = GitHawkApp

Reply:
Yep, that seems fair. Maybe create a `GitHawkApp` repo constant somewhere and check that. Didn't think of that.

Reply:
Ya I'll create that. 👍

Reply:
Could go fully overpowered and add it as an option to the markdown table at the top of template 😉but nah think the simple and easy solution is as above!

Reply:
Haha @Sherlouk if we added more details into the bug report that could be a cool way to allows other repos to make use of the general bug report info. But i don't know what useful data we could add to the bug report that benefits any repo.

Reply:
Switched checking titles to checking repo owner, now only GitHawkApp will have the bug report dump.